### PR TITLE
Don't crash when calling Tracing::Config methods without tracing configured

### PR DESCRIPTION
**Pull request by jwendell**
Fixes https://github.com/envoyproxy/envoy/issues/13164

Signed-off-by: Jonh Wendell <jonh.wendell@redhat.com>

### CODE REVIEW

**Comment by dmitri-d** [comment]
This approach looks good, could you add test(s) please?

File: source/common/http/conn_manager_impl.cc [hunk 1]
```diff
@@ -1533,19 +1533,39 @@ void ConnectionManagerImpl::ActiveStream::onBelowWriteBufferLowWatermark() {
 }
 
 Tracing::OperationName ConnectionManagerImpl::ActiveStream::operationName() const {
-  return connection_manager_.config_.tracingConfig()->operation_name_;
+  if (connection_manager_.config_.tracingConfig()) {
+    return connection_manager_.config_.tracingConfig()->operation_name_;
+  }
+
+  ENVOY_LOG(error, "tracingConfig()::operationName() called but tracing is not configured");
```
Comment (mattklein123) [inline source/common/http/conn_manager_impl.cc:1540]: For all of these, I would recommend returning an optional, and then letting the caller deal with ignoring or not.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks for working on this. A small comment to get started and per other comment also needs tests. Thank you!

/wait

File: source/common/http/conn_manager_impl.cc (hunk 1 above)
Comment (jwendell) [reply source/common/http/conn_manager_impl.cc:1540]: @mattklein123 This PR only fixes the crash. It doesn't attempt to change API.

I had done this in this another PR: https://github.com/envoyproxy/envoy/pull/13734. But it went too far in the sense it touched API and became too big. Therefore I thought about going small and only fix the crash issue with this PR and attack the other issue (i.e.: caller knows when tracing is configured or not) in a follow up. What do you think?

Comment (mattklein123) [reply source/common/http/conn_manager_impl.cc:1540]: I would rather just do the proper fix. I don't think it's that bad.

**Comment by jwendell** [comment]
Ok, closing this one in favor of my original PR.

**Pull request closed without merging**